File: eshoppers/src/main/java/com/shihabmahamud/eshoppers/util/Validations.java
package com.shihabmahamud.eshoppers.util;

import java.util.regex.Pattern;

public class Validations {
    Boolean email(String email) {
        if (email == null) return false;
        final var emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\." +
                "[a-zA-Z0-9_+&*-]+)*@" +
                "(?:[a-zA-Z0-9-]+\\.)+[a-z" +
                "A-Z]{2,7}$";
        final var pat = Pattern.compile(emailRegex);
        return pat.matcher(email).matches();
    }

    Boolean strLen(String str, Integer min, Integer max) {
        if (str.length() < min) return false;
        return str.length() <= max;
    }
}

File: eshoppers/src/main/kotlin/algos/Validations.kt
package algos

import java.util.regex.Pattern

object Validations {
    fun email(email: String?): Boolean {
        if (email == null) return false
        val emailRegex = "^[a-zA-Z0-9_+&*-]+(?:\\." +
                "[a-zA-Z0-9_+&*-]+)*@" +
                "(?:[a-zA-Z0-9-]+\\.)+[a-z" +
                "A-Z]{2,7}$"
        val pat = Pattern.compile(emailRegex)
        return pat.matcher(email).matches()
    }

    fun strLen(str: String, min: Int, max: Int): Boolean {
        if (str.length < min) return false
        return str.length <= max
    }
}